src/global/reco/reco.cc

```diff
@@ -166,6 +169,23 @@ void InitPlugin(JApplication *app) {
         app
     ));
 
+    app->Add(new JOmniFactoryGeneratorT<ReconstructedElectrons_factory>(
+        "ReconstructedElectronsForDIS",
+        {"MCParticles", "ReconstructedChargedParticles", "ReconstructedChargedParticleAssociations",
+        "EcalBarrelScFiClusterAssociations",
+        "EcalEndcapNClusterAssociations",
+        "EcalEndcapPClusterAssociations",
+        "EcalEndcapPInsertClusterAssociations",
+        "EcalLumiSpecClusterAssociations",
+        },
+        {"ReconstructedElectronsForDIS"},
+        {
+                                        .min_energy_over_momentum = 0.7, // GeV
+                                        .max_energy_over_momentum = 1.3  // GeV
+                                },
+        app
+    ));
```

Comment:
```suggestion
    
    auto EoverP = []( edm4eic::ReconstructedParticle p){
      if(p.getClusters.size()==0) return 0;
      double EoverP = p.getClusters()[0].getEnergy()/edm4hep::utils::magnitude(p.getMomentum());
      if(EoverP>0.7 && EoverP<1.3) return 1;
      else return 0;
      }
    
    app->Add(new JOmniFactoryGeneratorT<SubdivideCollection_factory>(
        "ReconstructedElectronsForDIS",
        {"ReconstructedParticles"
        },
        {"ReconstructedElectronsForDIS"},
        {
          .function = EoverPcut,
                                },
        app
    ));
```

Or something similar might work, I haven't tested yet. Someone else might be able to comment on the whether this is completely consistent with the current approach.

Reply:
Hi Simon, I think that makes sense to simplify things and avoid duplicating code between places.  Talking with @jdbrice when the first iterations of the electron ID were introduced, the MatchClusters algorithm wasn't quite ready yet, so he implemented his own version. 

However, if it is alright with you, what I would propose for now is to leave it as is for now.  Unless there is some technical issue with how it is right now, in which case I'm happy to work on it, we are nearing the end of the month and would like to have this pull request in for the April simulation campaign.  What I will do is open an issue to change the ReconstructedElectrons_factory to use the output of the MatchClusters_factory and try to get that done in time for the April campaign, but I want to avoid holding this up.

Reply:
Fine by me. There are a couple of other small changes that are needed to keep in line with the recent #1346

Reply:
Hi @simonge your solution is quite compact and elegant. However, in the future the electron ID (for DIS) is meant to utilize other info in addition to only the EoverP cut - so while your suggestion does replace the existing code, in the future we still want a more sophisticated approach for electron ID